engine/include/logger/log.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <logger/target.hpp>
+#include <source_location>
+#include <string_view>
+
+namespace gen::logger {
+using SrcLoc = std::source_location;
+
+void error(std::string_view category, std::string_view message, SrcLoc const& location = SrcLoc::current());
```

Comment:
minor opinion on the matter, but I'd say we should instead use the monikers: trace, log, warn, error, and fatal.

Though personally I'm a bit indifferent to trace and log though I do find them more explicit in their purpose for logging. As for fatal, I feel this is a much more important log type to allow for the filtering of fatal crashes. Though it can be argued that error can also take up this torch. Though I'd like to also hear your opinion on the matter.

Reply:
Fatal makes sense if it's different from error, which by your description it should be (by throwing an exception). Would asserts be discouraged in favour of `fatal` logs in that case? 

As for traces, I don't believe they are needed for basic client applications, as we can hook up a debugger / tracy / etc to the app at any time (unlike a remote app / server). It becomes useful when hard-to-reproduce Heisenbugs are involved, but it we get into that state we have much bigger problems to worry about anyway.

Reply:
PS: Will change it to `log`, the reason I prefer `info()` is because `log` is ambiguous. Also, `[E]`, `[W]`, `[I]` are recognized by many log readers / text editors and often coloured accordingly. Not sure about `[L]` and `[F]`.

Reply:
![Screenshot_20230716_011134](https://github.com/Rinzii/genesis/assets/16272243/3b832d46-2b84-4fe8-864c-536b6b5d8ed5)
Data point (VSCode)

Reply:
Hmmm, it's rough for me as I want something that is explicit like fatal but I don't think allowing fatal itself to throw exceptions is the right choice. I'd say then you are right that if fatal is not different from error then it does not make sense. 

I'll say then we won't add a fatal as I don't like the idea of allowing fatal to actually perform exceptions.

As for trace, you have a valid point. I'll say then it's fine to exclude it.